index.js

```diff
@@ -343,163 +341,159 @@ Filter.prototype._handleFile = function(relativePath, srcDir, destDir, entry, ou
-  // @ts-ignore
-  if (this.inputPaths === undefined) {
-    return false;
+  /**
+   * @public
+   *
+   * optionally override this to build a more rhobust cache key
+   * @param  {String} string The contents of a file that is being processed
+   * @return {String}        A cache key
+   */
+  cacheKeyProcessString (string, relativePath) {
+    return md5Hex(string + 0x00 + relativePath);
   }
 
-  // @ts-ignore
-  let srcDir = this.inputPaths[0];
-  let path = srcDir + '/' + relativePath;
+  canProcessFile(relativePath, entry) {
+    return !!this.getDestFilePath(relativePath, entry);
+  }
 
-  return (entry || fs.lstatSync(path)).isDirectory();
-};
+  isDirectory (relativePath, entry) {
+    // @ts-ignore
+    if (this.inputPaths === undefined) {
+      return false;
+    }
 
-Filter.prototype.getDestFilePath = function(relativePath, entry) {
-  // NOTE: relativePath may have been moved or unlinked
-  if (this.isDirectory(relativePath, entry)) {
-    return null;
-  }
+    // @ts-ignore
+    let srcDir = this.inputPaths[0];
+    let path = srcDir + '/' + relativePath;
 
-  if (this.extensions == null) {
-    return relativePath;
+    return (entry || fs.lstatSync(path)).isDirectory();
   }
 
-  for (let i = 0, ii = this.extensions.length; i < ii; ++i) {
-    let ext = this.extensions[i];
-    if (relativePath.slice(-ext.length - 1) === '.' + ext) {
-      if (this.targetExtension != null) {
-        relativePath = relativePath.slice(0, -ext.length) + this.targetExtension;
-      }
+  getDestFilePath (relativePath, entry) {
+    // NOTE: relativePath may have been moved or unlinked
+    if (this.isDirectory(relativePath, entry)) {
+      return null;
+    }
+
+    if (this.extensions == null) {
       return relativePath;
     }
+
+    for (let i = 0, ii = this.extensions.length; i < ii; ++i) {
+      let ext = this.extensions[i];
+      if (relativePath.slice(-ext.length - 1) === '.' + ext) {
+        if (this.targetExtension != null) {
+          relativePath = relativePath.slice(0, -ext.length) + this.targetExtension;
+        }
+        return relativePath;
+      }
+    }
+
+    return null;
   }
 
-  return null;
-};
-
-Filter.prototype.processAndCacheFile = function(srcDir, destDir, entry, forceInvalidation, isChange, instrumentation) {
-  let filter = this;
-  let relativePath = entry.relativePath;
-
-  return Promise.resolve().
-    then(() => {
-      return filter.processFile(srcDir, destDir, relativePath, forceInvalidation, isChange, instrumentation, entry);
-    }).
-    then(undefined,
-      // TODO(@caitp): error wrapper is for API compat, but is not particularly
-      // useful.
-      // istanbul ignore next
-      e => {
-        if (typeof e !== 'object') e = new Error('' + e);
-        e.file = relativePath;
-        e.treeDir = srcDir;
-        throw e;
-      });
-};
+  processAndCacheFile (srcDir, destDir, entry, forceInvalidation, isChange, instrumentation) {
```

Comment:
since we are dropping node 8, this function is also better written as an async function.